js/url_search_params_test.ts

```diff
@@ -22,6 +22,14 @@ test(function urlSearchParamsInitRecord(): void {
   assertEquals(searchParams.toString(), "a=54&b=true");
 });
 
+test(function urlSearchParamsInit(): void {
+  const params1 = new URLSearchParams("a=b");
+  assertEquals(params1.toString(), "a=b");
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const params2 = new URLSearchParams(params1 as any);
```

Comment:
Why does it need to be cast to `any`? Can we just add `URLSearchParams` to the list of constructor types accepted?

Reply:
`URLSearchParams` is implemented using typescript, but when we use deno in js code, developer can pass in any type.

`URLSearchParams` IDL Constructor: 

1. `sequence<sequence<USVString>>` or 
1. `record<USVString, USVString>` or 
1. `USVString`

but  in js code(not ts), we can pass in anything, such as `new URLSearchParams(0)`, `new URLSearchParams(true)`, ...

in wpt's [test case](https://github.com/web-platform-tests/wpt/blob/master/url/urlsearchparams-constructor.any.js#L1-L10), the param's type is `URLSearchParams` itself

```ts
var params = new URLSearchParams('a=b');
params = new URLSearchParams(params);
console.log(params.toString())
```

We expect to get `a=b`, but in Deno, we got `params=a%2Cb`.

Reply:
Ok